ibis/expr/rewrites.py

```diff
@@ -23,6 +28,55 @@
 name = var("name")
 
 
+class DerefMap(Concrete, Traversable):
```

Comment:
Can you add a docstring? This is a pretty important part of the library, and it would help folks understand how it fits into everything if there were some docs.